src/mds/Mutation.h

```diff
@@ -407,6 +411,8 @@ struct MDRequestImpl : public MutationImpl {
 
   // TrackedOp stuff
   typedef boost::intrusive_ptr<MDRequestImpl> Ref;
+  set<Ref> batch_reqs;
```

Comment:
`std::set` has awful cache locality and heap usage. I'd prefer we use a vector and `std::find` to check if the MDRequestRef already exists.